### PR TITLE
RHODS-8307:  Migrate biased onnx-model tests to CI

### PR DESCRIPTION
**Pull request by RobGeada**
Adds to CI:

- ONNX models
- DIR tests
- Multiple metric creation + deletion
- Varying metric values
- Larger data volumes

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: tests/basictests/trustyai_biased_models.sh [hunk 1]
```diff
@@ -162,50 +154,39 @@ function teardown_trustyai_test() {
...
 
+get_authentication
+check_trustyai_resources
+deploy_model
+check_mm_resources
+check_communication
+send_data
+schedule_and_check_request
+test_prometheus_scraping
+teardown_trustyai_test
```
Comment (danielezonca) [inline tests/basictests/trustyai_biased_models.sh:203]: Are the clusters used for CI where this is running stateful? For example what does it happen if there is an error and the teardown procedure fails (or it is skipped)?
If yes, I would try to make more fault tolerant the code, for example, is it possible to move this clean up step in a different block (similar to always() condition in GH action) to make sure that if any of the previous steps has crash, the cluster is not going to remain in a not consistent state?

While if the cluster is recreated/cleaned up automatically after the job, this teardown should not be necessary

File: e2e_tests/README.md [hunk 2]
```diff
@@ -1,40 +0,0 @@
-# E2E Tests
```
Comment (danielezonca) [inline e2e_tests/README.md:1]: I'm fine removing this copy of the manifests, but is it possible now to replicate CI behaviour/scripts locally? What about a readme file to describe how to do it?

**Review by danielezonca — COMMENTED** [review]
Thanks for the PR, nice cleanup and better CI :)
A couple of comments

**Review by ruivieira — APPROVED** [review]

File: e2e_tests/README.md (hunk 2 above)
Comment (RobGeada) [reply e2e_tests/README.md:1]: Yep, I'll add a readme

File: tests/README.md [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
-## Cleaning up after your test run (optional)
 
-Only run the following if you want to eliminate your Open Data Hub installation.
 
-To cleanup the Open Data Hub installation after a test run, you can run `make clean`.
-Running `make clean` **will wipe your Open Data Hub installation** and delete the project.
+## Useful Arguments
+* `BUILD_TOOL=docker/podman`: set the tool used to build and run the testing container
+* `SKIP_INSTALL=true/false`: skip the install of the ODH operator, if you've already installed it manually or via a previous test
+* `SKIP_KFDEF_INSTALL=true/false`: skip the install of ODH via KFdef, if you've already installed it manually or via a previous test
```
Comment (danielezonca) [inline tests/README.md:25]: Side note more for a future ticket, we should verify if this works with RHODS operator too. I guess so but it should be better to check it

File: tests/scripts/installandtest.sh [hunk 4]
```diff
@@ -26,7 +27,11 @@ if [ -z "${SKIP_INSTALL}" ]; then
     oc project ${ODHPROJECT} # in case a new project is not created
     $HOME/peak/install.sh
     echo "Sleeping for 5 min to let the KfDef install settle"
```
Comment (danielezonca) [inline tests/scripts/installandtest.sh]: Minor thing: you should move this into the `if else` because the duration of the sleep depends on `${LOCAL}`

**Review by danielezonca — APPROVED** [review]
Thanks for the PR, approving 👍

**Review by danielezonca — APPROVED** [review]
/lgtm

**Pull request merged**